spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/ZookeeperDiscoveryClientConfiguration.java

```diff
@@ -79,6 +76,7 @@ public ZookeeperDiscoveryClient zookeeperDiscoveryClient() {
 		private ZookeeperDependencies zookeeperDependencies;
 
 		@Bean
+		@ConditionalOnMissingBean
```

Comment:
Very good idea - IMO we should have as much `@ConditionalOnMissingBean` and `@ConditionalOnProperty` as possible